### ISSUE TITLE
Should be able to take the pager

### ISSUE THREAD

**Issue by adamgibbins** [PagerDuty/lita-pagerduty#8]
It'd be awesome if people could take the pager by being in Slack.
Obviously I'd expect this functionality to only work for people who have associated their PD email to their Lita username/handle.

lita: pager me 5m
Would give me the pager for 5 minutes by adding an override.

For reference, the commonly used PagerDuty Hubot plugin has this functionality.

**Comment by jaxxstorm** [comment]
The way the hubot plugin does this is post to the overrides API:

https://developer.pagerduty.com/documentation/rest/schedules/overrides/create

The main problem is that as part of the overrides API you need to specify a start and end time. The hubot plugin does this using moment:

```
start     = moment().format()
        minutes   = parseInt msg.match[3]
        end       = moment().add(minutes, 'minutes').format()
        override  = {
          'start':     start,
          'end':       end,
          'user_id':   userId
        }
```

Should be fairly easy to do in Ruby, I'll have a bash at this if I get time.

**Comment by jaxxstorm** [comment]
Probably also need a helper method to return a schedule from a string. the hubot plugin has a nice example here: https://github.com/hubot-scripts/hubot-pager-me/blob/acfafe03f687771d385c73b2c72ff08a3b0fe40b/src/scripts/pagerduty.coffee#L349-L376

**Comment by moos3** [comment]
Thanks. I am also working on this too and if I get it working. I'll make a pull request.

**Comment by jaxxstorm** [comment]
@moos3 I didn't get time to look at this yet, so hope you get something working!

I do think the pagerduty-sdk is a bit sparse. I think this library is better supported and more full featured: https://github.com/envato/pagerduty

**Comment by scarytom** [comment]
I also started having a look at this.  Indeed, the pagerduty-sdk gem does not support `POST schedules/:schedule_id/overrides` so it'll either need patching upstream, or lita-pagerduty will need to switch to another gem for communicating with the pagerduty API.

**Comment by scarytom** [comment]
I've submitted a pull request for pagerduty-sdk to add support for setting schedule overrides https://github.com/kryptek/pagerduty-sdk/pull/10

**Comment by tristaneuan** [comment]
Thanks for your contributions, @scarytom! This feature has been released in v0.2.0.
